include/albatross/src/core/distribution.hpp

```diff
@@ -113,6 +113,25 @@ void set_subset(const Distribution<CovarianceType> &from,
   }
 }
 
+inline MarginalDistribution
+concatenate_marginals(const MarginalDistribution &x,
+                      const MarginalDistribution &y) {
+  Eigen::VectorXd mean(x.mean.size() + y.mean.size());
+  mean.block(0, 0, x.mean.size(), 1) = x.mean;
+  mean.block(x.mean.size(), 0, y.mean.size(), 1) = y.mean;
+
+  Eigen::VectorXd variance = Eigen::VectorXd::Zero(mean.size());
```

Comment:
This logic looks weird to me. Do x and y have covariances if they are marginals? If only one of them has a covariance, do we get a concatenation with no uncertainty in some means but uncertainty in others? If neither has a covariance, is the concatenation a marginal with a diagonal of 0?

Reply:
yes, yes, yes :)

the `has_covariance` is sort of a poor name.  It's checking to make sure that an actual matrix has been allocated, in the case of a marginal distribution that "covariance" has zero off diagonals ... but it does exist.

Reply:
Would it be better to assert that there is a "covariance"?

Reply:
That would prevent us from being able to concatenate two marginal distributions if one of them doesn't have a covariance defined which seems a bit restrictive to me.

In general I'd like to just gid rid of the "optional" covariance for `MarginalDistribution` but it is used in a few places (namely `fit`) to allow you do pass targets which are the mean only through a model happily.  I'd be excited to change that behavior ... but that's a larger (less urgent) task.